dace/codegen/targets/xilinx.py

```diff
@@ -435,15 +439,39 @@ def generate_no_dependence_pre(kernel_stream,
                                    var_name=None):
         pass
 
-    def generate_no_dependence_post(self, kernel_stream, sdfg, state_id, node,
-                                    var_name):
+    def generate_no_dependence_post(
+            self,
+            kernel_stream,
+            sdfg: SDFG,
+            state_id: int,
+            node: nodes.Node,
+            var_name: str,
+            accessed_subset: Union[int, subsets.Subset] = None):
         '''
         Adds post loop pragma for ignoring loop carried dependencies on a given variable
         '''
         defined_type, _ = self._dispatcher.defined_vars.get(var_name)
-        if defined_type == DefinedType.ArrayInterface:
-            var_name = cpp.array_interface_variable(var_name, True,
-                                                    self._dispatcher)
+        newname = None
+
+        if var_name in sdfg.arrays:
+            array = sdfg.arrays[var_name]
+            if utils.is_HBM_array(array):
+                newname = cpp.ptr(var_name,
+                                  array,
+                                  accessed_subset,
+                                  sdfg,
+                                  True,
+                                  self._dispatcher,
+                                  is_array_interface=True)
+        if newname is not None:
+            newname = cpp.ptr(
+                var_name,
+                None,
+                None,
+                sdfg,
+                True,
+                self._dispatcher,
+                is_array_interface=(defined_type == DefinedType.ArrayInterface))
```

Comment:
Restructure this so you only have one call to `ptr`, and just set the arguments accordingly